ktor-client/ktor-client-tests/common/test/io/ktor/client/tests/HttpTimeoutTest.kt

```diff
@@ -0,0 +1,346 @@
+                response.readUTF8Line()
+            }
+        }
+    }
+
+    @Test
+    fun getRequestTimeoutWithSeparateReceivePerRequestAttributesTest() = clientTests(listOf("Curl", "Ios", "Js")) {
+        config {
+            install(HttpTimeout)
+        }
+
+        test { client ->
+            val response = client.request<ByteReadChannel>("$TEST_SERVER/timeout/with-stream?delay=500") {
+                method = HttpMethod.Get
+                setExtension(HttpTimeout.Extension.key, HttpTimeout.Extension(requestTimeout = 1000))
+            }
+            assertFailsWithRootCause<HttpRequestTimeoutException> {
+                response.readUTF8Line()
+            }
+        }
+    }
+
+    @Test
+    fun getStreamTest() = clientTests {
+        config {
+            install(HttpTimeout) { requestTimeout = 500 }
+        }
+
+        test { client ->
+            val response = client.get<ByteArray>("$TEST_SERVER/timeout/with-stream?delay=10")
+
+            assertEquals("Text", String(response))
+        }
+    }
+
+    @Test
+    fun getStreamPerRequestAttributesTest() = clientTests {
+        config {
+            install(HttpTimeout)
+        }
+
+        test { client ->
+            val response = client.get<ByteArray>("$TEST_SERVER/timeout/with-stream?delay=10") {
+                setExtension(HttpTimeout.Extension.key, HttpTimeout.Extension(requestTimeout = 500))
+            }
+
+            assertEquals("Text", String(response))
+        }
+    }
+
+    @Test
+    fun getStreamRequestTimeoutTest() = clientTests {
+        config {
+            install(HttpTimeout) { requestTimeout = 500 }
+        }
+
+        test { client ->
+            assertFailsWithRootCause<HttpRequestTimeoutException> {
+                client.get<ByteArray>("$TEST_SERVER/timeout/with-stream?delay=200")
+            }
+        }
+    }
+
+    @Test
+    fun getStreamRequestTimeoutPerRequestAttributesTest() = clientTests {
+        config {
+            install(HttpTimeout)
+        }
+
+        test { client ->
+            assertFailsWithRootCause<HttpRequestTimeoutException> {
+                client.get<ByteArray>("$TEST_SERVER/timeout/with-stream?delay=200") {
+                    setExtension(HttpTimeout.Extension.key, HttpTimeout.Extension(requestTimeout = 500))
+                }
+            }
+        }
+    }
+
+    @Test
+    fun redirectTest() = clientTests {
+        config {
+            install(HttpTimeout) { requestTimeout = 500 }
+        }
+
+        test { client ->
+            val response = client.get<String>("$TEST_SERVER/timeout/with-redirect?delay=10&count=2")
+            assertEquals("Text", response)
+        }
+    }
+
+    @Test
+    fun redirectPerRequestAttributesTest() = clientTests {
+        config {
+            install(HttpTimeout)
+        }
+
+        test { client ->
+            val response = client.get<String>("$TEST_SERVER/timeout/with-redirect?delay=10&count=2") {
```

Comment:
consider extracting url parameters in builder

Reply:
Done.